### PR TITLE
[double-conversion] Update to 3.2.1

### PR DESCRIPTION
**Pull request by Neumann-A**

### CODE REVIEW

**Comment by JonLiu1993** [comment]
@Neumann-A, Thanks for your pr, please get the ci error:
````
FAILED: bin/itkTestDriver.exe 
cmd.exe /C "cd . && D:\downloads\tools\cmake-3.25.0-windows\cmake-3.25.0-windows-i386\bin\cmake.exe -E vs_link_exe --intdir=Modules\Core\TestKernel\src\CMakeFiles\itkTestDriver.dir --rc=C:\PROGRA~2\WI3CF2~1\10\bin\100226~1.0\x64\rc.exe --mt=C:\PROGRA~2\WI3CF2~1\10\bin\100226~1.0\x64\mt.exe --manifests  -- C:\PROGRA~1\MICROS~1\2022\ENTERP~1\VC\Tools\MSVC\1434~1.319\bin\Hostx64\x64\link.exe  Modules\Core\TestKernel\src\CMakeFiles\itkTestDriver.dir\itkTestDriver.cxx.obj  /out:bin\itkTestDriver.exe /implib:lib\itkTestDriver.lib /pdb:bin\itkTestDriver.pdb /version:0.0 /machine:x64  /nologo    /debug /INCREMENTAL /subsystem:console -LIBPATH:D:\installed\x64-windows-static\lib lib\ITKIOBMP-5.3.lib  lib\ITKIOGDCM-5.3.lib  lib\ITKIOGIPL-5.3.lib  lib\ITKIOJPEG-5.3.lib  lib\ITKIOMeshBYU-5.3.lib  lib\ITKIOMeshFreeSurfer-5.3.lib  lib\ITKIOMeshGifti-5.3.lib  lib\ITKIOMeshOBJ-5.3.lib  lib\ITKIOMeshOFF-5.3.lib  lib\ITKIOMeshVTK-5.3.lib  lib\ITKIOMeta-5.3.lib  lib\ITKIONIFTI-5.3.lib  lib\ITKIONRRD-5.3.lib  lib\ITKIOPNG-5.3.lib  lib\ITKIOTIFF-5.3.lib  lib\ITKIOVTK-5.3.lib  lib\ITKCommon-5.3.lib  lib\ITKIOImageBase-5.3.lib  lib\ITKTestKernel-5.3.lib  lib\ITKIOBMP-5.3.lib  lib\ITKIOGDCM-5.3.lib  D:\installed\x64-windows-static\debug\lib\gdcmMSFF.lib  D:\installed\x64-windows-static\debug\lib\gdcmDICT.lib  D:\installed\x64-windows-static\debug\lib\gdcmIOD.lib  D:\installed\x64-windows-static\debug\lib\gdcmDSED.lib  D:\installed\x64-windows-static\debug\lib\gdcmCommon.lib  crypt32.lib  D:\installed\x64-windows-static\debug\lib\gdcmjpeg8.lib  D:\installed\x64-windows-static\debug\lib\gdcmjpeg12.lib  D:\installed\x64-windows-static\debug\lib\gdcmjpeg16.lib  D:\installed\x64-windows-static\debug\lib\openjp2.lib  D:\installed\x64-windows-static\debug\lib\gdcmcharls.lib  rpcrt4.lib  lib\ITKIOGIPL-5.3.lib  lib\ITKIOJPEG-5.3.lib  lib\ITKIOMeshBYU-5.3.lib  lib\ITKIOMeshFreeSurfer-5.3.lib  lib\ITKIOMeshGifti-5.3.lib  lib\ITKgiftiio-5.3.lib  D:\installed\x64-windows-static\debug\lib\libexpatdMT.lib  lib\ITKIOMeshOBJ-5.3.lib  lib\ITKIOMeshOFF-5.3.lib  lib\ITKIOMeshVTK-5.3.lib  lib\ITKIOMeshBase-5.3.lib  lib\ITKQuadEdgeMesh-5.3.lib  lib\ITKMesh-5.3.lib  lib\ITKIOMeta-5.3.lib  lib\ITKMetaIO-5.3.lib  comctl32.lib  wsock32.lib  lib\ITKIONIFTI-5.3.lib  lib\ITKniftiio-5.3.lib  lib\ITKznz-5.3.lib  lib\ITKTransform-5.3.lib  lib\ITKIONRRD-5.3.lib  lib\ITKNrrdIO-5.3.lib  lib\ITKIOPNG-5.3.lib  D:\installed\x64-windows-static\debug\lib\libpng16d.lib  lib\ITKIOTIFF-5.3.lib  D:\installed\x64-windows-static\debug\lib\tiffd.lib  D:\installed\x64-windows-static\debug\lib\lzma.lib  D:\installed\x64-windows-static\debug\lib\zlibd.lib  D:\installed\x64-windows-static\debug\lib\jpeg.lib  lib\ITKIOVTK-5.3.lib  lib\ITKIOImageBase-5.3.lib  lib\ITKCommon-5.3.lib  D:\installed\x64-windows-static\lib\double-conversion.lib  lib\itksys-5.3.lib  ws2_32.lib  dbghelp.lib  psapi.lib  lib\ITKVNLInstantiation-5.3.lib  lib\itkvnl_algo-5.3.lib  lib\itkvnl-5.3.lib  lib\itkv3p_netlib-5.3.lib  lib\itkvcl-5.3.lib  kernel32.lib user32.lib gdi32.lib winspool.lib shell32.lib ole32.lib oleaut32.lib uuid.lib comdlg32.lib advapi32.lib && cd ."
LINK Pass 1: command "C:\PROGRA~1\MICROS~1\2022\ENTERP~1\VC\Tools\MSVC\1434~1.319\bin\Hostx64\x64\link.exe Modules\Core\TestKernel\src\CMakeFiles\itkTestDriver.dir\itkTestDriver.cxx.obj /out:bin\itkTestDriver.exe /implib:lib\itkTestDriver.lib /pdb:bin\itkTestDriver.pdb /version:0.0 /machine:x64 /nologo /debug /INCREMENTAL /subsystem:console -LIBPATH:D:\installed\x64-windows-static\lib lib\ITKIOBMP-5.3.lib lib\ITKIOGDCM-5.3.lib lib\ITKIOGIPL-5.3.lib lib\ITKIOJPEG-5.3.lib lib\ITKIOMeshBYU-5.3.lib lib\ITKIOMeshFreeSurfer-5.3.lib lib\ITKIOMeshGifti-5.3.lib lib\ITKIOMeshOBJ-5.3.lib lib\ITKIOMeshOFF-5.3.lib lib\ITKIOMeshVTK-5.3.lib lib\ITKIOMeta-5.3.lib lib\ITKIONIFTI-5.3.lib lib\ITKIONRRD-5.3.lib lib\ITKIOPNG-5.3.lib lib\ITKIOTIFF-5.3.lib lib\ITKIOVTK-5.3.lib lib\ITKCommon-5.3.lib lib\ITKIOImageBase-5.3.lib lib\ITKTestKernel-5.3.lib lib\ITKIOBMP-5.3.lib lib\ITKIOGDCM-5.3.lib D:\installed\x64-windows-static\debug\lib\gdcmMSFF.lib D:\installed\x64-windows-static\debug\lib\gdcmDICT.lib D:\installed\x64-windows-static\debug\lib\gdcmIOD.lib D:\installed\x64-windows-static\debug\lib\gdcmDSED.lib D:\installed\x64-windows-static\debug\lib\gdcmCommon.lib crypt32.lib D:\installed\x64-windows-static\debug\lib\gdcmjpeg8.lib D:\installed\x64-windows-static\debug\lib\gdcmjpeg12.lib D:\installed\x64-windows-static\debug\lib\gdcmjpeg16.lib D:\installed\x64-windows-static\debug\lib\openjp2.lib D:\installed\x64-windows-static\debug\lib\gdcmcharls.lib rpcrt4.lib lib\ITKIOGIPL-5.3.lib lib\ITKIOJPEG-5.3.lib lib\ITKIOMeshBYU-5.3.lib lib\ITKIOMeshFreeSurfer-5.3.lib lib\ITKIOMeshGifti-5.3.lib lib\ITKgiftiio-5.3.lib D:\installed\x64-windows-static\debug\lib\libexpatdMT.lib lib\ITKIOMeshOBJ-5.3.lib lib\ITKIOMeshOFF-5.3.lib lib\ITKIOMeshVTK-5.3.lib lib\ITKIOMeshBase-5.3.lib lib\ITKQuadEdgeMesh-5.3.lib lib\ITKMesh-5.3.lib lib\ITKIOMeta-5.3.lib lib\ITKMetaIO-5.3.lib comctl32.lib wsock32.lib lib\ITKIONIFTI-5.3.lib lib\ITKniftiio-5.3.lib lib\ITKznz-5.3.lib lib\ITKTransform-5.3.lib lib\ITKIONRRD-5.3.lib lib\ITKNrrdIO-5.3.lib lib\ITKIOPNG-5.3.lib D:\installed\x64-windows-static\debug\lib\libpng16d.lib lib\ITKIOTIFF-5.3.lib D:\installed\x64-windows-static\debug\lib\tiffd.lib D:\installed\x64-windows-static\debug\lib\lzma.lib D:\installed\x64-windows-static\debug\lib\zlibd.lib D:\installed\x64-windows-static\debug\lib\jpeg.lib lib\ITKIOVTK-5.3.lib lib\ITKIOImageBase-5.3.lib lib\ITKCommon-5.3.lib D:\installed\x64-windows-static\lib\double-conversion.lib lib\itksys-5.3.lib ws2_32.lib dbghelp.lib psapi.lib lib\ITKVNLInstantiation-5.3.lib lib\itkvnl_algo-5.3.lib lib\itkvnl-5.3.lib lib\itkv3p_netlib-5.3.lib lib\itkvcl-5.3.lib kernel32.lib user32.lib gdi32.lib winspool.lib shell32.lib ole32.lib oleaut32.lib uuid.lib comdlg32.lib advapi32.lib /MANIFEST /MANIFESTFILE:Modules\Core\TestKernel\src\CMakeFiles\itkTestDriver.dir/intermediate.manifest Modules\Core\TestKernel\src\CMakeFiles\itkTestDriver.dir/manifest.res" failed (exit code 1319) with the following output:
double-conversion.lib(double-to-string.cc.obj) : error LNK2038: mismatch detected for '_ITERATOR_DEBUG_LEVEL': value '0' doesn't match value '2' in itkTestDriver.cxx.obj
double-conversion.lib(double-to-string.cc.obj) : error LNK2038: mismatch detected for 'RuntimeLibrary': value 'MT_StaticRelease' doesn't match value 'MTd_StaticDebug' in itkTestDriver.cxx.obj
double-conversion.lib(bignum-dtoa.cc.obj) : error LNK2038: mismatch detected for '_ITERATOR_DEBUG_LEVEL': value '0' doesn't match value '2' in itkTestDriver.cxx.obj
double-conversion.lib(bignum-dtoa.cc.obj) : error LNK2038: mismatch detected for 'RuntimeLibrary': value 'MT_StaticRelease' doesn't match value 'MTd_StaticDebug' in itkTestDriver.cxx.obj
double-conversion.lib(fixed-dtoa.cc.obj) : error LNK2038: mismatch detected for '_ITERATOR_DEBUG_LEVEL': value '0' doesn't match value '2' in itkTestDriver.cxx.obj
double-conversion.lib(fixed-dtoa.cc.obj) : error LNK2038: mismatch detected for 'RuntimeLibrary': value 'MT_StaticRelease' doesn't match value 'MTd_StaticDebug' in itkTestDriver.cxx.obj
double-conversion.lib(bignum.cc.obj) : error LNK2038: mismatch detected for '_ITERATOR_DEBUG_LEVEL': value '0' doesn't match value '2' in itkTestDriver.cxx.obj
double-conversion.lib(bignum.cc.obj) : error LNK2038: mismatch detected for 'RuntimeLibrary': value 'MT_StaticRelease' doesn't match value 'MTd_StaticDebug' in itkTestDriver.cxx.obj
double-conversion.lib(cached-powers.cc.obj) : error LNK2038: mismatch detected for '_ITERATOR_DEBUG_LEVEL': value '0' doesn't match value '2' in itkTestDriver.cxx.obj

double-conversion.lib(cached-powers.cc.obj) : error LNK2038: mismatch detected for 'RuntimeLibrary': value 'MT_StaticRelease' doesn't match value 'MTd_StaticDebug' in itkTestDriver.cxx.obj

LIBCMTD.lib(initializers.obj) : warning LNK4098: defaultlib 'libcmt.lib' conflicts with use of other libs; use /NODEFAULTLIB:library

bin\itkTestDriver.exe : fatal error LNK1319: 10 mismatches detected

ninja: build stopped: subcommand failed.
````

File: ports/itk/double-conversion.patch [hunk 1]
```diff
@@ -2,13 +2,19 @@ diff --git a/Modules/ThirdParty/DoubleConversion/CMakeLists.txt b/Modules/ThirdP
...
    get_target_property(ITKDoubleConversion_INCLUDE_DIRS double-conversion::double-conversion INTERFACE_INCLUDE_DIRECTORIES)
 -  get_target_property(ITKDoubleConversion_LIBRARIES double-conversion::double-conversion LOCATION)
-+  get_target_property(ITKDoubleConversion_LIBRARY_RELEASE double-conversion::double-conversion LOCATION_RELEASE)
-+  get_target_property(ITKDoubleConversion_LIBRARY_DEBUG double-conversion::double-conversion LOCATION_DEBUG)
++  get_target_property(ITKDoubleConversion_LIBRARY_RELEASE double-conversion::double-conversion IMPORTED_IMPLIB_RELEASE)
++  if(NOT ITKDoubleConversion_LIBRARY_RELEASE)
++    get_target_property(ITKDoubleConversion_LIBRARY_RELEASE double-conversion::double-conversion IMPORTED_LOCATION_RELEASE)
++  endif()
++  get_target_property(ITKDoubleConversion_LIBRARY_DEBUG double-conversion::double-conversion IMPORTED_IMPLIB_DEBUG)
++  if(NOT IMPORTED_IMPLIB_DEBUG)
```
Comment (Neumann-A) [inline ports/itk/double-conversion.patch:15]:
```suggestion
+  if(NOT ITKDoubleConversion_LIBRARY_DEBUG)
```

File: ports/itk/double-conversion.patch [hunk 2]
```diff
@@ -2,13 +2,19 @@ diff --git a/Modules/ThirdParty/DoubleConversion/CMakeLists.txt b/Modules/ThirdP
...
 -  get_target_property(ITKDoubleConversion_LIBRARIES double-conversion::double-conversion LOCATION)
-+  get_target_property(ITKDoubleConversion_LIBRARY_RELEASE double-conversion::double-conversion LOCATION_RELEASE)
-+  get_target_property(ITKDoubleConversion_LIBRARY_DEBUG double-conversion::double-conversion LOCATION_DEBUG)
++  get_target_property(ITKDoubleConversion_LIBRARY_RELEASE double-conversion::double-conversion IMPORTED_IMPLIB_RELEASE)
++  if(NOT ITKDoubleConversion_LIBRARY_RELEASE)
++    get_target_property(ITKDoubleConversion_LIBRARY_RELEASE double-conversion::double-conversion IMPORTED_LOCATION_RELEASE)
++  endif()
++  get_target_property(ITKDoubleConversion_LIBRARY_DEBUG double-conversion::double-conversion IMPORTED_IMPLIB_DEBUG)
++  if(NOT ITKDoubleConversion_LIBRARY_DEBUG)
++    get_target_property(ITKDoubleConversion_LIBRARY_DEBUG double-conversion::double-conversion IMPORTTED_LOCATION_DEBUG)
```
Comment (Neumann-A) [inline ports/itk/double-conversion.patch:16]:
```suggestion
+    get_target_property(ITKDoubleConversion_LIBRARY_DEBUG double-conversion::double-conversion IMPORTED_LOCATION_DEBUG)
```

**Comment by Neumann-A** [comment]
@JonLiu1993 PR is green.

**Review by JonLiu1993 — APPROVED** [review]

**Pull request merged**